docs(adr): adding the adr process documentation (#120)
allinbits/cosmos-cash: docs/architecture/README.md

# Architecture Decision Records (ADR)

> Use this location to record all high-level architecture decisions in the Cosmos SDK.

Author: ```suggestion
Use this location to record all high-level architecture decisions in the Cosmos Cash.
```

# Architecture Decision Records (ADR)

> This is a location to record all high-level architecture decisions in the Cosmos-SDK.

Reviewer: ```suggestion
The location to record all high-level architecture decisions in the Cosmos SDK.
```

> This is a location to record all high-level architecture decisions in the Cosmos-SDK.

Reviewer: oh!! Cosmos-Cash it is? interesting

> 

Reviewer: @SenorMonito did you know about the GitHub suggesting feature? Looks like you were suggesting here, but your sentence landed outside the triple tick marks, so I moved the line inside, per
![image](https://user-images.githubusercontent.com/10067215/124780416-92c84000-df10-11eb-84d9-491678dbb36d.png)

An Architectural Decision (**AD**) is a software design choice that addresses a functional or non-functional requirement that is architecturally significant.
An Architecturally Significant Requirement (**ASR**) is a requirement that has a measurable effect on a software system’s architecture and quality.

> An Architectural Decision Record (**ADR**) captures a single AD, such as often done when writing personal notes or meeting minutes; the collection of ADRs created and maintained in a project constitute its decision log. All these are within the topic of Architectural Knowledge Management (AKM).

Reviewer: ```suggestion
An Architectural Decision Record (**ADR**) captures a single AD, and is as often done when writing personal notes or meeting minutes. The collection of ADRs created and maintained in a project constitute its decision log. All these records are within the topic of Architectural Knowledge Management (AKM).
```

> You can read more about the ADR concept in this [blog post](https://product.reverb.com/documenting-architecture-decisions-the-reverb-way-a3563bb24bd0#.78xhdix6t).

Reviewer: ```suggestion
You can read more about the ADR concept in the [Documenting architecture decisions, the Reverb way](https://product.reverb.com/documenting-architecture-decisions-the-reverb-way-a3563bb24bd0#.78xhdix6t) blog post.
```

Reviewer: @PaddyMc we have non-descriptive link text all over our technical content, but our readers benefit from using descriptive link text https://developers.google.com/style/link-text

Reviewer: @SenorMonito you can see I suggested changes to the link text that include the blog article title... we have so many links that say [here] which are literally not helpful. (hi David!)

## Rationale

ADRs are intended to be the primary mechanism for proposing new feature designs and new processes, for collecting community input on an issue, and for documenting the design decisions.
An ADR should provide:

- Context on the relevant goals and the current state
- Proposed changes to achieve the goals
- Summary of pros and cons
- References
- Changelog

Note the distinction between an ADR and a spec. The ADR provides the context, intuition, reasoning, and
justification for a change in architecture, or for the architecture of something
new. The spec is much more compressed and streamlined summary of everything as
it stands today.

> If recorded decisions turned out to be lacking, convene a discussion, record the new decisions here, and then modify the code to match.

Reviewer: ```suggestion
If recorded decisions turned out to be lacking the required substance, the process is to convene a discussion, record the new decisions here, and then modify the code to match.
```
